### PR TITLE
feat(module:cron-expression): add cron-expression component

### PR DESCRIPTION
**Pull request by OriginRing**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: components/i18n/nz-i18n.interface.ts [hunk 1]
```diff
@@ -131,6 +131,25 @@ export interface NzTextI18nInterface {
   expand: string;
 }
 
+export interface NzCronExpressionI18nInterface {
+  cronMore1: string;
+  cronMore2: string;
+  cronMore3: string;
```
Comment (simplejason) [inline components/i18n/nz-i18n.interface.ts:137]: 1、2、3 这种命名不要有，看下有没有语义化的描述

**Review by simplejason — CHANGES_REQUESTED** [review]

File: components/components.less [hunk 2]
```diff
@@ -62,3 +62,4 @@
 @import './result/style/entry.less';
 @import './space/style/entry.less';
 @import './image/style/entry.less';
+@import './cron-expression/style/entry';
```
Comment (hsuanxyz) [inline components/components.less:65]:
```suggestion
@import './cron-expression/style/entry.less';
```

File: components/cron-expression/cron-expression-specialized.component.ts [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+      <form nz-form [nzLayout]="'inline'" [formGroup]="validateForm">
+        <div class="ant-cron-expression-box">
+          <div class="ant-cron-expression-content">
+            <div
+              class="ant-cron-expression-input-group"
+              [class.ant-cron-expression-input-group-lg]="nzSize === 'large'"
+              [class.ant-cron-expression-input-group-sm]="nzSize === 'small'"
+            >
+              <div class="ant-cron-expression-input">
+                <input nz-input formControlName="minute" id="minute" name="minute" />
```
Comment (hsuanxyz) [inline components/cron-expression/cron-expression-specialized.component.ts:52]: Are there `id`(s) here required? If so, use the dynamic `id`.

File: components/cron-expression/cron-expression-specialized.component.ts [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+    hour: ['*', Validators.required, this.hour],
+    day: ['*', Validators.required, this.day],
+    month: ['*', Validators.required, this.month],
+    week: ['?', Validators.required, this.week]
+  });
+
+  constructor(private formBuilder: UntypedFormBuilder, private cdr: ChangeDetectorRef, private i18n: NzI18nService) {}
+
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+  onChange: any = () => {};
```
Comment (hsuanxyz) [inline components/cron-expression/cron-expression-specialized.component.ts:269]:
```suggestion
  onChange: SafeAny = () => {};
```

File: components/cron-expression/cron-expression-specialized.component.ts [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+  onTouch: () => void = () => null;
+
+  writeValue(value: CronType): void {
+    this.validateForm.patchValue(value);
+  }
+
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+  registerOnChange(fn: any): void {
```
Comment (hsuanxyz) [inline components/cron-expression/cron-expression-specialized.component.ts:277]:
```suggestion
  registerOnChange(fn: SafeAny): void {
```

File: components/cron-expression/cron-expression-specialized.component.ts [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+    this.onChange = fn;
+  }
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+  registerOnTouched(fn: any): void {
```
Comment (hsuanxyz) [inline components/cron-expression/cron-expression-specialized.component.ts:281]:
```suggestion
  registerOnTouched(fn: SafeAny): void {
```

File: components/cron-expression/cron-expression.component.ts [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+  @Input() nzType: NzButtonType = 'default';
+  @Input() nzDefaultConfigure: CronSettings = [];
+  private destroy$ = new Subject<void>();
+
+  validateForm: UntypedFormGroup = this.formBuilder.group({
+    specialized: { minute: '*', hour: '*', day: '*', month: '*', week: '*' }
+  });
+
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+  onChange: any = () => {};
```
Comment (hsuanxyz) [inline components/cron-expression/cron-expression.component.ts:69]:
```suggestion
  onChange: SafeAny = () => {};
```

File: components/cron-expression/cron-expression.component.ts [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+  registerOnChange(fn: any): void {
+    this.onChange = fn;
+  }
+
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+  registerOnTouched(fn: any): void {
+    this.onTouch = fn;
+  }
```
Comment (hsuanxyz) [inline components/cron-expression/cron-expression.component.ts:92]:
```suggestion
  registerOnChange(fn: SafeAny): void {
    this.onChange = fn;
  }

  registerOnTouched(fn: SafeAny): void {
    this.onTouch = fn;
  }
```

**Review by hsuanxyz — CHANGES_REQUESTED** [review]

**Review by hsuanxyz — APPROVED** [review]
LGTM

File: components/cron-expression/cron-expression.component.ts [hunk 9]
```diff
@@ -0,0 +1,265 @@
...
+        </div>
+        <nz-collapse *ngIf="!nzCollapseDisable" [nzBordered]="false">
+          <nz-collapse-panel [nzHeader]="nextDate">
+            <ng-container *ngIf="validateForm.valid">
+              <div class="ant-cron-expression-collapse-date">
+                <p style="margin: 0" *ngFor="let dateItem of nextTimeList">
+                  {{ dateItem | date: 'YYYY-MM-dd HH:mm:ss' }}
+                </p>
+              </div>
+              <a (click)="loadDate()">···</a>
```
Comment (hsuanxyz) [inline components/cron-expression/cron-expression.component.ts:81]:
```suggestion
              <ul class="ant-cron-expression-preview">
                <li *ngFor="let dateItem of nextTimeList">
                  {{ dateItem | date: 'YYYY-MM-dd HH:mm:ss' }}
                </li>
                 <li><a (click)="loadDate()">···</a></li>
              </ul>
             
```

**Review by simplejason — APPROVED** [review]
LGTM

**Pull request merged**